flipperdevices/flipperzero-firmware
# Fix Text Input: overwrite max size template

# What's new

- The TextInput would not allow another key press if the template that is
going to be overwritten is over the over the buffer size.

- This change clears any default text, before checking the length of
entered input.

# Verification 

To see the issue, use the verification method without the PR applied

- You would need a plugin or application that uses a TextInput and edit its code
- Configure the maxlength of the TextInput to a known value
- Configure a template text (`clear_default_text = 1`) that is exactly this length
- Compile and run the plugin/app
- Without this PR:
  - See that the TextInput does not allow a new text to be entered, unless Backspace is used
- With this PR applied:
  - Verify that the TextInput now allows any character to overwrite the template text

# Checklist (For Reviewer)
